obsreport/obsreport_receiver.go

```diff
@@ -63,7 +87,70 @@ func NewReceiver(cfg ReceiverSettings) *Receiver {
 			tag.Upsert(obsmetrics.TagKeyTransport, cfg.Transport, tag.WithTTL(tag.TTLNoPropagation)),
 		},
 		tracer: cfg.ReceiverCreateSettings.TracerProvider.Tracer(cfg.ReceiverID.String()),
+		meter:  cfg.ReceiverCreateSettings.MeterProvider.Meter(cfg.ReceiverID.String()),
+		logger: cfg.ReceiverCreateSettings.Logger,
+
+		otelEnabled: InstrumentWithOtel(),
+		otelAttrs: []attribute.KeyValue{
+			attrReceiver.String(cfg.ReceiverID.String()),
+			attrTransport.String(cfg.Transport),
+		},
+	}
+
+	rec.createOtelMetrics()
+
+	return rec
+}
+
+func (rec *Receiver) createOtelMetrics() {
+	var err error
+	handleError := func(metricName string, err error) {
+		if err != nil {
+			rec.logger.Warn("failed to create otel instrument", zap.Error(err), zap.String("metric", metricName))
+		}
 	}
+
+	rec.acceptedSpansCounter, err = rec.meter.SyncInt64().Counter(
```

Comment:
thought: I wonder if these should exist in a list/map that are then just accessed as needed. i.e.

```
        telemetryMap = map[config.DataType]map[string]*syncint64.Counter
        // iterate through [DataType] x ["accepted", "refused"] to initialize
...
	telemetryMap[dataType]["accepted"].Add(receiverCtx, int64(numAccepted), rec.otelAttrs...)
	telemetryMap[dataType]["refused"].Add(receiverCtx, int64(numRefused), rec.otelAttrs...)
```

Reply:
Accessing a map is slower than direct access. So I prefer this way.